dbms/src/Flash/Mpp/ExchangeReceiver.cpp

```diff
@@ -865,14 +755,15 @@ void ExchangeReceiverBase<RPCContext>::connectionDone(
         copy_live_conn);
 
     if (copy_live_conn == 0)
-    {
         LOG_DEBUG(log, "All threads end in ExchangeReceiver");
-    }
     else if (copy_live_conn < 0)
         throw Exception("live_connections should not be less than 0!");
 
     if (meet_error || copy_live_conn == 0)
+    {
         finishAllMsgChannels();
+        cv.notify_all();
```

Comment:
> Why not call `cv.notify_all()` in L758?

OK, I think call it in L758 is better otherwise `cv.notify_all()` will be called unnecessarily when meet_error is true.